tests/model_notebooks/test_notebooks.py

```diff
@@ -0,0 +1,361 @@
+        for notebook_path in self.notebooks_found:
+            success = self._execute_notebook(notebook_path)
+            if success:
+                self.successful_notebooks.append(notebook_path)
+
+        # Log summary
+        total_count = len(self.notebooks_found)
+        success_count = len(self.successful_notebooks)
+        fail_count = len(self.failed_notebooks)
+
+        logger.info("=" * 60)
+        logger.info("Test run complete")
+        logger.info(f"{success_count}/{total_count} notebook(s) passed")
+
+        if self.failed_notebooks:
+            failed_names = [nb.name for nb in self.failed_notebooks.keys()]
+            logger.error(f"{fail_count} notebook(s) failed: {', '.join(failed_names)}")
+
+            # Raise assertion error to fail the test
+            raise AssertionError(
+                f"{fail_count} notebook(s) failed to execute: {', '.join(failed_names)}"
+            )
+        else:
+            logger.success("All notebooks passed!")
+
+    def get_summary(self) -> Dict[str, Any]:
+        """
+        Get a summary of the test run.
+
+        Returns:
+            Dictionary containing test run metadata:
+                - total_notebooks: Total number of notebooks found
+                - successful_notebooks: Number of successful executions
+                - failed_notebooks: Number of failed executions
+                - failed_notebook_names: List of failed notebook names
+                - notebook_directory: Path to notebook directory
+                - results_directory: Path to results directory
+        """
+        return {
+            "total_notebooks": len(self.notebooks_found),
+            "successful_notebooks": len(self.successful_notebooks),
+            "failed_notebooks": len(self.failed_notebooks),
+            "failed_notebook_names": [nb.name for nb in self.failed_notebooks.keys()],
+            "notebook_directory": str(self.notebook_directory),
+            "model_directory": str(self.model_dir),
+        }
+
+
+# Pytest test functions
+@pytest.mark.slow
+def test_interactive_notebooks() -> None:
+    """Test all notebooks in the interactive directory."""
+    # Skip test if not on SLURM
+    if not IS_ON_SLURM:
+        pytest.skip("Test skipped: must be run on SLURM cluster")
+
+    runner = NotebookTestRunner(
+        notebook_directory="tests/model_notebooks/interactive",
+        environment_type="simulation",
+    )
+    runner.test_run_notebooks()
+
+
+@pytest.mark.slow
+def test_results_notebooks() -> None:
+    """Test all notebooks in the results directory."""
+    # Skip test if not on SLURM
+    if not IS_ON_SLURM:
+        pytest.skip("Test skipped: must be run on SLURM cluster")
+
+    runner = NotebookTestRunner(
+        notebook_directory="tests/model_notebooks/results",
+        environment_type="artifact",
+    )
+    runner.test_run_notebooks()
+
+
+@pytest.mark.slow
+@pytest.mark.skip(reason="No notebooks currently in artifact directory")
```

Comment:
This currently causes an error to be thrown but there are plans to add more notebooks in the near future.